aya/src/programs/lirc_mode2.rs

```diff
@@ -93,6 +151,17 @@ impl LircLink {
             target_fd: Some(unsafe { dup(target_fd) }),
         }
     }
+
+    pub fn info(&self) -> Result<bpf_prog_info, ProgramError> {
+        if let Some(fd) = self.prog_fd {
```

Comment:
I'm not sure how I feel about exposing `bpf_prog_info` as part of the API. What's your use case? I'm thinking we should maybe return some kind of wrapper type.

Reply:
The bpf_prog_info struct has a lot of fields and a lot of them are simple `u64` fields, so the wrapper will be pretty thin.

So we could have a `struct ProgramInfo(bpf_prog_info)` with a load of accessor functions. Would that work?

Reply:
That sounds good yes, and for now I would implement only the accessors that you need. If people need more accessors they'll open an issue or send a PR, we'll learn about more use cases, and see if there's a better API we can provide.

Reply:
Great, thanks. I've done that, let me know what you think

Reply:
Looks good thanks!